src/games/strategy/triplea/ai/proAI/ProCombatMoveAI.java

```diff
@@ -427,23 +445,30 @@ private void determineTerritoriesToAttack(final Map<Territory, ProAttackTerritor
     ProLogger.debug("Final number of territories: " + (numToAttack - 1));
   }
 
-  private void determineTerritoriesThatCanBeHeld(final List<ProAttackTerritoryData> prioritizedTerritories,
-      final Map<Territory, ProAttackTerritoryData> attackMap,
-      final Map<Territory, ProAttackTerritoryData> enemyAttackMap, final Map<Territory, Double> territoryValueMap) {
+  private void determineTerritoriesThatCanBeHeld(final List<ProTerritory> prioritizedTerritories,
+      final Map<Territory, ProTerritory> attackMap, final ProMoveOptions enemyAttackOptions,
+      final Map<Territory, Double> territoryValueMap) {
 
     ProLogger.info("Check if we should try to hold attack territories");
 
     // Determine which territories to try and hold
     final Map<Unit, Territory> unitTerritoryMap = utils.createUnitTerritoryMap(player);
-    for (final ProAttackTerritoryData patd : prioritizedTerritories) {
+    for (final ProTerritory patd : prioritizedTerritories) {
       final Territory t = patd.getTerritory();
 
+      // If strafing then can't hold
+      if (patd.isStrafing()) {
```

Comment:
Yeah. It seems that everyone's definition of 'strafing' is slightly different. I really consider it any attack that is meant to not capture a territory but weaken the enemy in some way (TUV swing, weaken territory defense, etc). You seem to have a different definition and some on the forum have different definitions again.

Its not really 'unblocking' since that to me would be clearing a territory so that you or an ally could move through it. It is really 'attacking to weaken defenses so an ally can capture' at least in this case. Not sure what the best term for that is but suggestions are welcome :)